libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +161,501 @@ export class DynamicFormGroup<TModel> extends FormGroup {
-                ((control as FormArray).controls[i] as FormControl).markAsTouched({ onlySelf: true });
-              } else {
-                if ((control as FormArray).controls[i] instanceof DynamicFormGroup) {
-                  ((control as FormArray).controls[i] as DynamicFormGroup<any>).validateAllFormFields();
+}
+
+
+export function getClassValidators<TModel>(factoryModel: ClassType<TModel>, fields: Dictionary, validatorOptions?: ValidatorOptions) {
+
+  // Get the validation rules from the object decorators
+  const allValidationMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(factoryModel, '');
+
+  // Get the validation rules for the validation group: https://github.com/typestack/class-validator#validation-groups
+  const validationGroupMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(
+      factoryModel,
+      '',
+      validatorOptions && validatorOptions.groups ? validatorOptions.groups : undefined
+    );
+
+  const formGroupFields = {};
+  const validator = new Validator();
+
+  // Loop through all fields in the form definition
+  Object.keys(fields).filter(key => key.indexOf('__') !== 0).forEach(fieldName => {
+
+    // Conditional Validation for the field
+    const conditionalValidations: ValidationMetadata[] = [];
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.conditional.type)) {
+        conditionalValidations.push(validationMetadata);
+      }
+    });
+
+    // All Nested Validation for the field
+    const allNestedValidations: ValidationMetadata[] = [];
+      allValidationMetadatas.forEach(validationMetadata => {
+        if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.nested.type)) {
+          allNestedValidations.push(validationMetadata);
+        }
+      });
+
+    // Nested Validation for the field for the requested class-validator group
+    const nestedGroupValidations: ValidationMetadata[] = [];
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (isPropertyValidatorOfType(validationMetadata, fieldName, ValidationKeys.nested.type)) {
+        nestedGroupValidations.push(validationMetadata);
+      }
+    });
+
+    const fieldDefinition: DynamicFormGroupField = {
+      data: formGroupFields[fieldName],
+      validation: []
+    };
+
+    if (fieldDefinition.data === undefined) {
+      fieldDefinition.data = fields[fieldName];
+    }
+
+    validationGroupMetadatas.forEach(validationMetadata => {
+      if (validationMetadata.propertyName === fieldName && validationMetadata.type !== ValidationKeys.conditional.type) {
+
+        // tslint:disable-next-line:forin
+        for (const typeKey in ValidationTypes) {
+
+          const canValidateField = ValidationTypes.hasOwnProperty(typeKey)
+            && validationMetadata.type === ValidationTypes[typeKey]
```

Comment:
Moved 
`validationMetadata.type === ValidationTypes[typeKey] && validator[validationMetadata.type] === undefined;` 

out of each individual check and put it at the top since it is a common condition in all checks